### PR TITLE
Property flags and descriptors

### PR DESCRIPTION
**Pull request by mahdiHashemi14**
Please review and comment/commit if anything is wrong

### CODE REVIEW

File: 1-js/07-object-properties/01-property-descriptors/article.md [hunk 1]
```diff
@@ -116,25 +116,25 @@ Object.defineProperty(user, "name", {
...
 */!*
 ```
 
-Now no one can change the name of our user, unless they apply their own `defineProperty` to override ours.
+حالا هیچ‌کس نمی‌تواند اسم کاربر ما را تغییر دهد، مگر اینکه `defineProperty` خودش را برای باطل کردن توصیف‌کننده‌ی ما اعمال کند.
 
-```smart header="Errors appear only in strict mode"
-In the non-strict mode, no errors occur when writing to non-writable properties and such. But the operation still won't succeed. Flag-violating actions are just silently ignored in non-strict.
+```smart header="ارورها فقط در حالت سخت‌گیرانه ایجاد می‌شوند"
+در حالت غیر سخت‌گیرانه، زمانی که ویژگی‌های غیر قابل نوشتن را تغییر و یا کاری مشابه انجام می‌دهیم اروری ایجاد نمی‌شود. اما همچنان این کار انجام نمی‌شود. در حالت غیر سخت‌گیرانه، کارهای نقص‌کننده‌ی پرچم بی سر و صدا نادیده گرفته می‌شوند.
```
Comment (mahdiHashemi14) [inline 1-js/07-object-properties/01-property-descriptors/article.md:126]: Line 126 needs more attention

File: 1-js/07-object-properties/01-property-descriptors/article.md [hunk 2]
```diff
@@ -259,24 +259,24 @@ Object.defineProperty(user, "name", {
...
+// یا پرچم‌های آن را تغییر دهیم user.name نمی‌توانیم
+// :هیچ کدام این‌ها کار نخواهند کرد
 user.name = "Pete";
 delete user.name;
 Object.defineProperty(user, "name", { value: "Pete" });
 ```
 
-```smart header="The only attribute change possible: writable true -> false"
-There's a minor exception about changing flags.
+```smart header="تنها تغییر ممکن روی صفت: writable true -> false"
```
Comment (mahdiHashemi14) [inline 1-js/07-object-properties/01-property-descriptors/article.md:269]: This header (line 269) needs more attention

File: 1-js/07-object-properties/01-property-descriptors/article.md [hunk 3]
```diff
@@ -296,54 +296,54 @@ Object.defineProperties(user, {
...
+پس ما می‌توانیم چند ویژگی را یک‌باره تنظیم کنیم.
 
-## Object.getOwnPropertyDescriptors
+## متد Object.getOwnPropertyDescriptors
 
-To get all property descriptors at once, we can use the method [Object.getOwnPropertyDescriptors(obj)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors).
+برای گرفتن تمام توصیف‌کننده‌های ویژگی با هم، می‌توانیم از متد [Object.getOwnPropertyDescriptors(obj)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors) استفاده کنیم.
 
-Together with `Object.defineProperties` it can be used as a "flags-aware" way of cloning an object:
+این متد همراه با `Object.defineProperties` می‌تواند به عنوان راهی «آگاه از پرچم‌ها» برای کپی کردن یک شیء استفاده شود:
```
Comment (mahdiHashemi14) [inline 1-js/07-object-properties/01-property-descriptors/article.md:305]: Line 305 needs more attention

File: 1-js/07-object-properties/01-property-descriptors/article.md (hunk 1 above)
Comment (mahdyar) [reply 1-js/07-object-properties/01-property-descriptors/article.md:126]:
```suggestion
در حالت غیر سخت‌گیرانه، زمانی که بر روی ویژگی‌های غیرقابل نوشتن می‌نویسیم، هیچ اروری رخ نمی‌دهد. اما همچنان این کار انجام نمی‌شود. در حالت غیر سخت‌گیرانه، کارهای نقص‌کننده‌ی پرچم بی سر و صدا نادیده گرفته می‌شوند.
```

File: 1-js/07-object-properties/01-property-descriptors/article.md (hunk 3 above)
Comment (mahdyar) [reply 1-js/07-object-properties/01-property-descriptors/article.md:305]:
```suggestion
این متد همراه با `Object.defineProperties` می‌تواند به عنوان راهی «همراه با پرچم‌ها» برای کپی کردن یک شیء استفاده شود:
```

**Review by mahdyar — CHANGES_REQUESTED** [review]
Please apply the suggestions.

**Comment by mahdiHashemi14** [comment]
Thank you @mahdyar 
/done

**Review by mahdyar — APPROVED** [review]

**Pull request merged**